cookie.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * @typedef {{
+ *      domain: string,
+ *      path: string,
+ *      secure: boolean,
+ *      ?expires: number|Date,
+ * }} mojave.CookieOptions
+ */
+
+
+/**
+ * Sets a cookie's value. If the value is an object or array, it'll be JSON.stringify and saved as a string
+ *
+ * @param {string} key
+ * @param {*} value
+ * @param {mojave.CookieOptions} options
+ */
+export function setCookie (key, value, options = {})
+{
+    options = Object.assign({
+        path: "/",
+        secure: window.location.protocol === "https",
+    }, options);
+
+    if (typeof options.expires === "number")
+    {
+        options.expires = new Date(new Date() * 1 + options.expires * 864e+5);
+    }
+
+    options.expires = typeof options.expires !== "undefined"
+        ? options.expires.toUTCString()
+        : "";
+
+    const encodedKey = encodeCookieKey(key);
+    const encodedValue = encodeCookieValue(value);
+    const encodedOptions = encodeCookieOptions(options);
+
+    document.cookie = `${encodedKey}=${encodedValue};${encodedOptions.join(" ;")}`;
+}
+
+
+/**
+ * Retrieves a cookie's value
+ *
+ * @param {string} key
+ *
+ * @returns {*|null}
+ */
+export function getCookie (key)
+{
+    const cookies = document.cookie.split("; ");
+
+    for (let i = 0; i < cookies.length; i++)
+    {
+        const parts = cookies[i].split("=");
+        const cookieName = decodeURIComponent(parts[0]);
+
+        let cookieValue = parts.slice(1).join("=");
+        if (cookieValue.charAt(0) === '"')
+        {
+            cookieValue = cookieValue.slice(1, -1);
+        }
+
+        const decodedCookieValue = decodeURIComponent(cookieValue);
+
+        if (cookieName === key)
```

Comment:
Why are we decoding everything for all cookies, although we maybe are only using the last one?

Reply:
I don't want to cache the cookies since it may have changed since the last call. If you can come up with a faster method I'm open for changes.

Reply:
This is more or less on-par with what Js-Cookie is doing. Pretty sure there might be some magic RegEx that could ease the entire parsing process, I just couldn't come up with one just yet.

Reply:
Yes, there is, but that was not what my comment was about.

I would just have added a 

```js
if (cookieName !== name)
{
    continue;
}
```

at the top of the loop, so that we don't do any work for cookies that are not interesting for us.

Reply:
Ah well :) It became obsolete with your suggested RegEx as soon as it passes my tests :)